### PR TITLE
Got time then do else ignore :Change AddressBook references to ContactList

### PR DESCRIPTION
**Pull request by rajobasu**

### CODE REVIEW

File: src/test/java/seedu/address/testutil/ContactListBuilder.java [hunk 1]
```diff
@@ -6,24 +6,24 @@
...
 
     private AddressBook addressBook;
 
-    public AddressBookBuilder() {
+    public ContactListBuilder() {
         addressBook = new AddressBook();
     }
 
-    public AddressBookBuilder(AddressBook addressBook) {
+    public ContactListBuilder(AddressBook addressBook) {
```
Comment (Shizheng001) [inline src/test/java/seedu/address/testutil/ContactListBuilder.java:19]: lets change parameters too

File: src/test/java/seedu/address/testutil/ContactListBuilder.java [hunk 2]
```diff
@@ -6,24 +6,24 @@
 /**
  * A utility class to help with building Addressbook objects.
  * Example usage: <br>
- *     {@code AddressBook ab = new AddressBookBuilder().withCustomer("John", "Doe").build();}
+ *     {@code AddressBook ab = new ContactListBuilder().withCustomer("John", "Doe").build();}
  */
-public class AddressBookBuilder {
+public class ContactListBuilder {
 
     private AddressBook addressBook;
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/testutil/ContactListBuilder.java]: Class is wrong and variable name is wrong

File: src/test/java/seedu/address/testutil/ContactListBuilder.java [hunk 3]
```diff
@@ -6,24 +6,24 @@
 /**
  * A utility class to help with building Addressbook objects.
  * Example usage: <br>
- *     {@code AddressBook ab = new AddressBookBuilder().withCustomer("John", "Doe").build();}
+ *     {@code AddressBook ab = new ContactListBuilder().withCustomer("John", "Doe").build();}
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/testutil/ContactListBuilder.java]: `AddressBook ab` should be `ContactList cl`

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 4]
```diff
@@ -51,8 +51,8 @@ public void prefsReadSave() throws Exception {
     public void addressBookReadSave() throws Exception {
         /*
          * Note: This is an integration test that verifies the StorageManager is properly wired to the
-         * {@link JsonAddressBookStorage} class.
-         * More extensive testing of UserPref saving/reading is done in {@link JsonAddressBookStorageTest} class.
+         * {@link JsonContactListStorage} class.
+         * More extensive testing of UserPref saving/reading is done in {@link JsonContactListStorageTest} class.
          */
         AddressBook original = getTypicalAddressBook();
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/storage/StorageManagerTest.java]: `getTypicalAddressBook` should be `getTypicalContactList`

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 5]
```diff
@@ -24,7 +24,7 @@
 
     @BeforeEach
     public void setUp() {
-        JsonAddressBookStorage addressBookStorage = new JsonAddressBookStorage(getTempFilePath("ab"));
+        JsonContactListStorage addressBookStorage = new JsonContactListStorage(getTempFilePath("ab"));
         JsonUserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(getTempFilePath("prefs"));
         storageManager = new StorageManager(addressBookStorage, userPrefsStorage);
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/storage/StorageManagerTest.java:29]: `addressBookStorage` should be `contactListStorage`

File: src/test/java/seedu/address/storage/JsonContactListStorageTest.java [hunk 6]
```diff
@@ -96,7 +96,7 @@ public void saveAddressBook_nullAddressBook_throwsNullPointerException() {
      */
     private void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) {
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/storage/JsonContactListStorageTest.java]: Method name again

File: src/test/java/seedu/address/storage/JsonContactListStorageTest.java [hunk 7]
```diff
@@ -47,41 +47,41 @@ public void read_missingFile_emptyResult() throws Exception {
...
+        jsonContactListStorage.saveAddressBook(original, filePath);
+        readBack = jsonContactListStorage.readAddressBook(filePath).get();
         assertEquals(original, new AddressBook(readBack));
 
         // Save and read without specifying file path
         original.addCustomer(IDA);
-        jsonAddressBookStorage.saveAddressBook(original); // file path not specified
-        readBack = jsonAddressBookStorage.readAddressBook().get(); // file path not specified
+        jsonContactListStorage.saveAddressBook(original); // file path not specified
+        readBack = jsonContactListStorage.readAddressBook().get(); // file path not specified
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/storage/JsonContactListStorageTest.java:84]: `readAddressBook` method is wrong

File: src/test/java/seedu/address/storage/JsonContactListStorageTest.java [hunk 8]
```diff
@@ -47,41 +47,41 @@ public void read_missingFile_emptyResult() throws Exception {
...
     }
 
     @Test
     public void readAddressBook_invalidAndValidCustomerAddressBook_throwDataConversionException() {
-        assertThrows(DataConversionException.class, () -> readAddressBook("invalidAndValidCustomerAddressBook.json"));
+        assertThrows(DataConversionException.class, () -> readAddressBook("invalidAndValidCustomerContactList.json"));
     }
 
     @Test
     public void readAndSaveAddressBook_allInOrder_success() throws Exception {
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/storage/JsonContactListStorageTest.java:64]: method name is wrong and probably wrong for many test case code

File: src/test/java/seedu/address/model/ModelManagerTest.java [hunk 9]
```diff
@@ -95,7 +95,7 @@ public void getFilteredCustomerList_modifyList_throwsUnsupportedOperationExcepti
 
     @Test
     public void equals() {
-        AddressBook addressBook = new AddressBookBuilder().withCustomer(ALICE).withCustomer(BENSON).build();
+        AddressBook addressBook = new ContactListBuilder().withCustomer(ALICE).withCustomer(BENSON).build();
         AddressBook differentAddressBook = new AddressBook();
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/model/ModelManagerTest.java]: variable name `differentAddressBook `

File: src/main/java/seedu/address/storage/JsonSerializableContactList.java [hunk 10]
```diff
@@ -17,26 +17,26 @@
...
+     * Constructs a {@code JsonSerializableContactList} with the given customers.
      */
     @JsonCreator
-    public JsonSerializableAddressBook(@JsonProperty("customers") List<JsonAdaptedCustomer> customers) {
+    public JsonSerializableContactList(@JsonProperty("customers") List<JsonAdaptedCustomer> customers) {
         this.customers.addAll(customers);
     }
 
     /**
      * Converts a given {@code ReadOnlyAddressBook} into this class for Jackson use.
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/storage/JsonSerializableContactList.java]: comment has reference to `ReadOnlyAddressBook`

File: src/main/java/seedu/address/storage/JsonSerializableContactList.java [hunk 11]
```diff
@@ -17,26 +17,26 @@
  * An Immutable AddressBook that is serializable to JSON format.
  */
 @JsonRootName(value = "addressbook")
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/storage/JsonSerializableContactList.java:19]: `value = "addressbook"` is probably wrong :(

File: src/main/java/seedu/address/storage/JsonContactListStorage.java [hunk 12]
```diff
@@ -74,7 +74,7 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) thro
         requireNonNull(filePath);
 
         FileUtil.createIfMissing(filePath);
-        JsonUtil.saveJsonFile(new JsonSerializableAddressBook(addressBook), filePath);
+        JsonUtil.saveJsonFile(new JsonSerializableContactList(addressBook), filePath);
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/storage/JsonContactListStorage.java:77]: `addressBook` variable name

File: src/main/java/seedu/address/storage/JsonContactListStorage.java [hunk 13]
```diff
@@ -17,13 +17,13 @@
 /**
  * A class to access AddressBook data stored as a json file on the hard disk.
  */
-public class JsonAddressBookStorage implements AddressBookStorage {
+public class JsonContactListStorage implements AddressBookStorage {
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/storage/JsonContactListStorage.java]: Class name `AddressBookStorage`

File: src/main/java/seedu/address/storage/JsonContactListStorage.java [hunk 14]
```diff
@@ -17,13 +17,13 @@
 /**
  * A class to access AddressBook data stored as a json file on the hard disk.
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/storage/JsonContactListStorage.java]: Comment is unchanged

File: src/main/java/seedu/address/storage/ContactListStorage.java [hunk 15]
```diff
@@ -5,41 +5,42 @@
...
      * @throws IOException if there was any problem when reading from the storage.
      */
-    Optional<ReadOnlyAddressBook> readAddressBook() throws DataConversionException, IOException;
+    Optional<ReadOnlyContactList> readAddressBook() throws DataConversionException, IOException;
 
     /**
      * @see #getAddressBookFilePath()
      */
-    Optional<ReadOnlyAddressBook> readAddressBook(Path filePath) throws DataConversionException, IOException;
+    Optional<ReadOnlyContactList> readAddressBook(Path filePath) throws DataConversionException, IOException;
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/storage/ContactListStorage.java:32]: method name

File: src/main/java/seedu/address/storage/ContactListStorage.java [hunk 16]
```diff
@@ -5,41 +5,42 @@
...
      */
-    void saveAddressBook(ReadOnlyAddressBook addressBook) throws IOException;
+    void saveAddressBook(ReadOnlyContactList addressBook) throws IOException;
 
     /**
-     * @see #saveAddressBook(ReadOnlyAddressBook)
+     * @see #saveAddressBook(ReadOnlyContactList)
      */
-    void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) throws IOException;
+    void saveAddressBook(ReadOnlyContactList addressBook, Path filePath) throws IOException;
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/storage/ContactListStorage.java:44]: variable name

Comment (nighoggDatatype) [reply src/main/java/seedu/address/storage/ContactListStorage.java:44]: Also method name

File: src/main/java/seedu/address/MainApp.java [hunk 17]
```diff
@@ -74,20 +74,20 @@ public void init() throws Exception {
      * or an empty address book will be used instead if errors occur when reading {@code storage}'s address book.
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/MainApp.java:74]: bad comment

File: src/main/java/seedu/address/MainApp.java [hunk 18]
```diff
@@ -74,20 +74,20 @@ public void init() throws Exception {
...
      */
     private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
-        Optional<ReadOnlyAddressBook> addressBookOptional;
-        ReadOnlyAddressBook initialData;
+        Optional<ReadOnlyContactList> addressBookOptional;
+        ReadOnlyContactList initialData;
         try {
             addressBookOptional = storage.readAddressBook();
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/MainApp.java:80]: variable name `addressBookOptional` and method name `readAddressBook`

File: src/main/java/seedu/address/MainApp.java [hunk 19]
```diff
@@ -74,20 +74,20 @@ public void init() throws Exception {
...
             if (!addressBookOptional.isPresent()) {
-                logger.info("Data file not found. Will be starting with a sample AddressBook");
+                logger.info("Data file not found. Will be starting with a sample ContactList");
             }
             initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/MainApp.java:84]: method `getSampleAddressBook`

File: src/test/java/seedu/address/testutil/TypicalCustomers.java [hunk 20]
```diff
@@ -60,10 +60,10 @@
     private TypicalCustomers() {} // prevents instantiation
 
     /**
-     * Returns an {@code AddressBook} with all the typical customers.
+     * Returns an {@code ContactList} with all the typical customers.
      */
-    public static AddressBook getTypicalAddressBook() {
-        AddressBook ab = new AddressBook();
+    public static ContactList getTypicalAddressBook() {
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/testutil/TypicalCustomers.java:65]: method name

File: src/test/java/seedu/address/testutil/ContactListBuilder.java [hunk 21]
```diff
@@ -0,0 +1,34 @@
+package seedu.address.testutil;
+
+import seedu.address.model.ContactList;
+import seedu.address.model.customer.Customer;
+
+/**
+ * A utility class to help with building Addressbook objects.
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/testutil/ContactListBuilder.java:7]: comment is bad

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 22]
```diff
@@ -24,7 +24,7 @@
 
     @BeforeEach
     public void setUp() {
-        JsonAddressBookStorage addressBookStorage = new JsonAddressBookStorage(getTempFilePath("ab"));
+        JsonContactListStorage addressBookStorage = new JsonContactListStorage(getTempFilePath("ab"));
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/storage/StorageManagerTest.java:27]: variable name `addressBookStorage `

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 23]
```diff
@@ -51,13 +51,13 @@ public void prefsReadSave() throws Exception {
     public void addressBookReadSave() throws Exception {
         /*
          * Note: This is an integration test that verifies the StorageManager is properly wired to the
-         * {@link JsonAddressBookStorage} class.
-         * More extensive testing of UserPref saving/reading is done in {@link JsonAddressBookStorageTest} class.
+         * {@link JsonContactListStorage} class.
+         * More extensive testing of UserPref saving/reading is done in {@link JsonContactListStorageTest} class.
          */
-        AddressBook original = getTypicalAddressBook();
+        ContactList original = getTypicalAddressBook();
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/storage/StorageManagerTest.java:57]: `getTypicalAddressBook`

File: src/test/java/seedu/address/storage/JsonContactListStorageTest.java [hunk 24]
```diff
@@ -105,6 +105,6 @@ private void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) {
 
     @Test
     public void saveAddressBook_nullFilePath_throwsNullPointerException() {
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/storage/JsonContactListStorageTest.java:107]: method name

File: src/test/java/seedu/address/storage/JsonContactListStorageTest.java [hunk 25]
```diff
@@ -105,6 +105,6 @@ private void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) {
...
-        assertThrows(NullPointerException.class, () -> saveAddressBook(new AddressBook(), null));
+        assertThrows(NullPointerException.class, () -> saveAddressBook(new ContactList(), null));
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/storage/JsonContactListStorageTest.java:108]: `saveAddressBook` is bad

File: src/test/java/seedu/address/model/ModelManagerTest.java [hunk 26]
```diff
@@ -113,20 +113,20 @@ public void equals() {
...
         modelManager.updateFilteredCustomerList(new NameContainsKeywordsPredicate(Arrays.asList(keywords)));
-        assertFalse(modelManager.equals(new ModelManager(addressBook, userPrefs)));
+        assertFalse(modelManager.equals(new ModelManager(contactList, userPrefs)));
 
         // resets modelManager to initial state for upcoming tests
         modelManager.updateFilteredCustomerList(PREDICATE_SHOW_ALL_CUSTOMERS);
 
         // different userPrefs -> returns false
         UserPrefs differentUserPrefs = new UserPrefs();
         differentUserPrefs.setAddressBookFilePath(Paths.get("differentFilePath"));
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/model/ModelManagerTest.java:129]: `setAddressBookFilePath` 👎

File: src/test/java/seedu/address/model/ContactListTest.java [hunk 27]
```diff
@@ -22,25 +22,25 @@
...
     }
 
     @Test
     public void resetData_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> addressBook.resetData(null));
+        assertThrows(NullPointerException.class, () -> contactList.resetData(null));
     }
 
     @Test
     public void resetData_withValidReadOnlyAddressBook_replacesData() {
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/model/ContactListTest.java:40]: method name

File: src/test/java/seedu/address/model/ContactListTest.java [hunk 28]
```diff
@@ -22,25 +22,25 @@
...
-        AddressBook newData = getTypicalAddressBook();
-        addressBook.resetData(newData);
-        assertEquals(newData, addressBook);
+        ContactList newData = getTypicalAddressBook();
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/model/ContactListTest.java:41]: `getTypicalAddressBook` is a thorn

File: src/test/java/seedu/address/logic/commands/ClearCommandTest.java [hunk 29]
```diff
@@ -24,7 +24,7 @@ public void execute_emptyAddressBook_success() {
     public void execute_nonEmptyAddressBook_success() {
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/logic/commands/ClearCommandTest.java:24]: method name 👎

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 30]
```diff
@@ -149,13 +149,13 @@ private void assertCommandFailure(String inputCommand, Class<? extends Throwable
...
-    private static class JsonAddressBookIoExceptionThrowingStub extends JsonAddressBookStorage {
-        private JsonAddressBookIoExceptionThrowingStub(Path filePath) {
+    private static class JsonContactListIoExceptionThrowingStub extends JsonContactListStorage {
+        private JsonContactListIoExceptionThrowingStub(Path filePath) {
             super(filePath);
         }
 
         @Override
-        public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) throws IOException {
+        public void saveAddressBook(ReadOnlyContactList addressBook, Path filePath) throws IOException {
```
Comment (nighoggDatatype) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:158]: method name and variable name 👎

File: src/main/java/seedu/address/ui/CustomerCard.java [hunk 31]
```diff
@@ -23,7 +23,7 @@
      * Note: Certain keywords such as "location" and "resources" are reserved keywords in JavaFX. As a consequence, UI
      * elements' variable names cannot be set to such keywords or an exception will be thrown by JavaFX during runtime.
      *
-     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on AddressBook level 4</a>
+     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on ContactList level 4</a>
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/ui/CustomerCard.java:26]: I think this one and this one alone should be unchanged given that it references an external project

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 32]
```diff
@@ -47,33 +47,33 @@ public void saveUserPrefs(ReadOnlyUserPrefs userPrefs) throws IOException {
...
     @Override
-    public void saveAddressBook(ReadOnlyAddressBook addressBook) throws IOException {
-        saveAddressBook(addressBook, addressBookStorage.getAddressBookFilePath());
+    public void saveAddressBook(ReadOnlyContactList addressBook) throws IOException {
+        saveAddressBook(addressBook, contactListStorage.getAddressBookFilePath());
     }
 
     @Override
-    public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) throws IOException {
+    public void saveAddressBook(ReadOnlyContactList addressBook, Path filePath) throws IOException {
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/storage/StorageManager.java:74]: More bad method names and variable names

File: src/main/java/seedu/address/storage/JsonSerializableContactList.java [hunk 33]
```diff
@@ -9,52 +9,52 @@
...
+     * @param source future changes to this will not affect the created {@code JsonSerializableContactList}.
      */
-    public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
+    public JsonSerializableContactList(ReadOnlyContactList source) {
         customers.addAll(source.getCustomerList().stream().map(JsonAdaptedCustomer::new).collect(Collectors.toList()));
     }
 
     /**
-     * Converts this address book into the model's {@code AddressBook} object.
+     * Converts this address book into the model's {@code ContactList} object.
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/storage/JsonSerializableContactList.java:44]: comment "Converts this **_address book_** into the model"

**Review by nighoggDatatype — CHANGES_REQUESTED** [review]
I am finishing my review now since I don't think I can be exhaustive and timely. What I advise is to do a global search for the word address, the word book, the phrase "address book" and "addressbook", all case insensitive searches. I don't think further comments would be helpful

**Pull request closed without merging**